src/red.ts

```diff
@@ -129,36 +131,55 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
         if (typeof blue === 'function') {
             return getRedFunction(blue);
         }
-        let isBlueArray = false;
-        try {
-            isBlueArray = isArrayOrNotOrThrowForRevoked(blue);
-        } catch {
-            // blue was revoked - but we call createRedProxy to support distortions
-            return createRedProxy(blue);
-        }
-        if (isBlueArray) {
-            return getRedArray(blue);
-        } else if (typeof blue === 'object') {
+        if (typeof blue === 'object') {
             const red: RedValue | undefined = WeakMapGet(blueMap, blue);
-            if (isUndefined(red)) {
-                return createRedProxy(blue);
+            if (!isUndefined(red)) {
+                return red;
+            }
+            let hasDynamicMark: boolean = false;
+            try {
+                hasDynamicMark = hasOwnPropertyCall(blue, LockerLiveValueMarkerSymbol);
+            } catch {}
```

Comment:
I think this entire portion can be extracted to a utility called `hasDynamicMark(value: any)` to shield the code from knowing about the symbol. We use the same code `getRedFunction`

Reply:
will do

Reply:
It would also be great to export a utility called "markDynamicObject" along with `createSecureEnvironment`. This way users will never have to worry about the internals of how dynamic objects are created and it allows for non breaking changes as code evolves.

Reply:
Folks will find out about the symbol (lwc will use it). The obscurity security doesn't apply here.